src/AspNetCore.Lambda/HttpHandlers.fs

```diff
@@ -275,4 +291,25 @@ let htmlFile (relativeFilePath : string) =
                 ctx
                 |> (setHttpHeader "Content-Type" "text/html"
                 >>= setBodyAsString html)
-        }
\ No newline at end of file
+        }
+
+let private handlerWithRootedPath (path:string) (handler : HttpHandler) = 
+    fun (ctx : HttpHandlerContext) ->
+        async {
+            let savedSubPath = getSavedSubPath ctx.HttpContext
+            try
+                ctx.HttpContext.Items.Item RouteKey <- ((savedSubPath |> Option.defaultValue "") + path)
+                return! handler ctx
+            finally
+                match savedSubPath with
+                | Some savedSubPath -> ctx.HttpContext.Items.Item RouteKey <- savedSubPath
+                | None              -> ctx.HttpContext.Items.Remove RouteKey |> ignore
+        }
+
```

Comment:
Let's add a description to be complete (feel free to re-phrase it you can come up with sth better).

/// Filters an incoming HTTP request based on the a part of the request path (case sensitive).
/// Subsequent route handlers inside the given handler function should omit the already validated subPath.